pkg/daemon/update.go

```diff
@@ -372,6 +372,14 @@ func (dn *Daemon) applyOSChanges(oldConfig, newConfig *mcfgv1.MachineConfig) (re
 		}
 	}
 
+	// apply Override Image
+	if mcDiff.overrideImage {
```

Comment:
Ideally this function should be called after updateOS() so that anything applied or removed is depsolved by rpm-ostree on latest content